ui/js/controllers/logviewer.js

```diff
@@ -269,10 +274,16 @@ logViewerApp.controller('LogviewerCtrl', [
                             if ($scope.step_data.all_errors.length == 0) {
                                 angular.element(document).ready(function () {
                                     if (isNaN($scope.selectedBegin)) {
-                                        $scope.displayLog($scope.step_data.steps[0], 'initialLoad');
-                                    } else {
-                                        moveScrollToLineNumber($scope.selectedBegin, $event);
+                                        for (var i = 0; i < $scope.step_data.steps.length; i++) {
+                                            var step = $scope.step_data.steps[i];
+                                            if (step.result !== "success") {
+                                                $scope.selectedBegin = step.finished_linenumber;
+                                                $scope.selectedEnd = step.finished_linenumber;
+                                                break;
+                                            }
+                                        }
                                     }
+                                    moveScrollToLineNumber($scope.selectedBegin, $event);
                                 });
                             } else {
                                 $timeout(function() {
```

Comment:
With setting the line number, this code now does some odd things when you DO have an error line.  It selects the right line, but the URL shows #L0 (at least it did for me in a few cases).
I think this entire ``if..else`` clause should be removed here and changed to:

    $scope.setLineNumber($scope.step_data.all_errors[0].linenumber);
    moveScrollToLineNumber($scope.selectedBegin, $event);

Since we've determined that ``$scope.step_data.all_errors.length`` is greater than 0, we can rely that we have at least one.  We take the first, get the line number and go with that.  No need to trigger a click.